lamindb_setup/core/_hub_core.py

```diff
@@ -439,32 +438,45 @@ def _access_aws(*, storage_root: str, client: Client) -> dict[str, dict]:
     return storage_root_info
 
 
-def access_db(isettings: InstanceSettings, access_token: str | None = None) -> str:
+def access_db(
+    instance: InstanceSettings | dict, access_token: str | None = None
+) -> str:
+    instance_id: UUID
+    instance_slug: str
+    instance_api_url: str | None
+    if isinstance(instance, InstanceSettings):
+        instance_id = instance._id
+        instance_slug = instance.slug
+        instance_api_url = instance._api_url
+    else:
+        instance_id = UUID(instance["id"])
+        instance_slug = instance["owner"] + "/" + instance["name"]
+        instance_api_url = instance["api_url"]
```

Comment:
The 9 line preceding lines here are not pretty at all. Why do we have to accept a `dict` here rather than the properly typed `InstanceSettings` object?

Reply:
It is needed in `lamindb` in `using`. It is better than to initialize `isettings` from the `dict`.